### PR TITLE
Closes #569  | Add/Update Dataloader MSL4Emergency

### LINKED ISSUE #569: Create dataset loader for Myanmar Sign Language Corpus for the Emergency Domain (MSL4Emergency)
Dataloader name: `msl4emergency/msl4emergency.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?msl4emergency

|     Dataset| msl4emergency |
|-------------|---|
| Description | The MSL4Emergency corpus is part of a larger Myanmar sign language (MSL) corpus that specifically contains sign language videos for the emergency domain. Each signing video is annotated with both its transcription and its Burmese written translation, which may differ from each other due to grammar, syntax and vocabulary differences between MSL and Burmese. Signing videos were made by sign language trainers and deaf trainees. |
| Subsets | - |
| Languages | ysm, mya |
| Tasks | Machine Translation, Sign Language Recognition |
| License | Creative Commons Attribution Non Commercial Share Alike 4.0 (cc-by-nc-sa-4.0) |
| Homepage | https://github.com/ye-kyaw-thu/MSL4Emergency/tree/master |
| HF URL | - |
| Paper URL | https://core.ac.uk/reader/489828410 |

### PR DESCRIPTION
**Pull request by sabilmakbar**
Please name your PR title and the first line of PR message after the issue it will close. You can use the following examples:

Closes #569 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/{my_dataset}/{my_dataset}.py` (please use only lowercase and underscore for dataset folder naming, as mentioned in dataset issue) and its `__init__.py` within `{my_dataset}` folder.
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_LOCAL`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py` or `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py --subset_id {subset_name_without_source_or_seacrowd_suffix}`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Review by muhammadravi251001 — CHANGES_REQUESTED** [review]
Somehow, I still get errors when running `.load_dataset` or testing via `tests.test_seacrowd`, the errors are:

`UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 11: character maps to <undefined>`.

Do you know why this happens? Or is everything fine (OK/passed) on ​​your side? @sabilmakbar

**Comment by sabilmakbar** [comment]
> Somehow, I still get error when running .load_dataset or testing via tests.test_seacrowd, the error is:
> 
> `UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 11: character maps to <undefined>.`
> 
> Do you know why this happens? Or is everything fine (OK/passed) on ​​your side? @sabilmakbar

Hmm, on my end it works fine. You can check this log file, which is a sh output from this cmd:
`python -m tests.test_seacrowd seacrowd/sea_datasets/msl4emergency/msl4emergency.py`

[test_msl4emergency.log](https://github.com/SEACrowd/seacrowd-datahub/files/15402373/test_msl4emergency.log)

I'm thinking if your error comes from these lines (in L114-115):
```python
with open(text_path, "r") as f:
    text_data = [data.split("\t") for data in f.readlines()]
```

You can try to print the `text_path` var and load it separately on your end to see if that's indeed the case.

**Comment by muhammadravi251001** [comment]
It's not about `_CITATION`. After changing the `encoding` parameter on `open` to:
```
with open(text_path, "r", encoding="utf-8") as f:
```
I get an OK response.

The problem is: that I run this dataloader implementation on my Windows notebook, and Windows do not always run encoding on `utf-8` in my case; while Linux and MacOS default is `utf-8`.

To avoid this future-encoding-error, I recommend adding an `encoding` value to the `open` method.

**Comment by sabilmakbar** [comment]
thanks! done making changes, pls have a look @muhammadravi251001

**Review by muhammadravi251001 — APPROVED** [review]
LGTM. Approved. Great work!

Thanks for the change & your contribution @sabilmakbar!

**Comment by holylovenia** [comment]
Hi @ljvmiranda921, sorry for the sudden notice. I assigned you here in place of @luckysusanto due to the lack of response.

I would like to let you know that we plan to finalize the calculation of the open contributions (e.g., dataloader implementations) **in 31 hours**, so it'd be great if we could wrap up the reviewing and merge this PR before then.

File: seacrowd/sea_datasets/msl4emergency/msl4emergency.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+_URL = "https://github.com/ye-kyaw-thu/MSL4Emergency/archive/refs/heads/master.zip"
+
+_SUPPORTED_TASKS = [Tasks.MACHINE_TRANSLATION, Tasks.SIGN_LANGUAGE_RECOGNITION]
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+_CONFIG_SUFFIXES_FOR_TASK = [TASK_TO_SCHEMA.get(task).lower() for task in _SUPPORTED_TASKS]
+
+
+class MSL4Emergency(datasets.GeneratorBasedBuilder):
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/msl4emergency/msl4emergency.py:50]:
```suggestion
class MSL4EmergencyDataset(datasets.GeneratorBasedBuilder):
```

**Comment by ljvmiranda921** [comment]
Sure! I'll just update the small nits myself as well just to speed things up! 👍

**Review by ljvmiranda921 — APPROVED** [review]
LGTM! Merging this in a few

**Pull request merged**